Sources/SwiftSyntax/SyntaxConvenienceMethods.swift

```diff
@@ -0,0 +1,80 @@
+//===- SyntaxConvenienceMethods.swift - Convenience funcs for syntax nodes ===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public extension FloatLiteralExprSyntax {
+  var floatingValue: Double {
+    return potentialFloatingValue!
+  }
+
+  fileprivate var potentialFloatingValue: Double? {
+    let floatingDigitsWithoutUnderscores = floatingDigits.text.filter {
+      $0 != "_"
+    }
+    return Double(floatingDigitsWithoutUnderscores)
+  }
+}
+
+public extension IntegerLiteralExprSyntax {
+  var integerValue: Int {
+    return potentialIntegerValue!
+  }
+
+  fileprivate var potentialIntegerValue: Int? {
+    let text = digits.text
+    let (prefixLength, radix) = IntegerLiteralExprSyntax.prefixLengthAndRadix(text: text)
+    let digitsStartIndex = text.index(text.startIndex, offsetBy: prefixLength)
+    let textWithoutPrefix = text.suffix(from: digitsStartIndex)
+
+    let textWithoutPrefixOrUnderscores = textWithoutPrefix.filter {
+      $0 != "_"
+    }
+
+    return Int(textWithoutPrefixOrUnderscores, radix: radix)
```

Comment:
I'm sorry to have missed the initial review, but this would crash for valid integer literals that are too large to be represented in `Int`. Is this intentional? If so, shouldn't this API be `intValue` and the floating-point API `doubleValue` and not `integerValue`/`floatingValue` (the latter implying that any valid integer or floating-point value would be returned)?

Reply:
According to the `Int` initializer's [documentation](https://github.com/apple/swift/blob/103961a7d5f8853ac518f4d8151de0f5b64d8ccf/stdlib/public/core/IntegerParsing.swift#L136), "if the value [that the `String` parameter] denotes in the given `radix` is not representable, the result is `nil`." I confirmed this by running the following code:
```
  let goodInt = Int("42", radix: 10)
  let badInt = Int("42424242424242424242424242424242424242424242424242424242424242424242424242", radix: 10)
  print("@@@ \(goodInt) @@@ \(badInt) @@@")
```
The output was:
```
@@@ Optional(42) @@@ nil @@@
```

Reply:
Right, so force unwrapping that value will crash. Is this the intended behavior here?

Reply:
A valid `IntegerLiteralExprSyntax` would not cause a crash.

The only way a crash would occur, via force unwrap, is if a client called `integerValue` on an invalid `IntegerLiteralExprSyntax`. However, there are guardrails in place to prevent clients from creating invalid `IntegerLiteralExprSyntax`es. Those guardrails, not initially present in the PR, were the primary focus of feedback.

Reply:
I might be misunderstanding then. My assumption was that 'IntegerLiteralExprSyntax' can represent any valid integer literal? If so, then calling 'integerValue' on some valid 'IntegerLiteralExprSyntax' values would crash, because not all valid integer literals can be represented as an 'Int'.

Reply:
`IntegerLiteralExprSyntax`es represent `Int` literals in source files. A non-representable `Int` won't compile.

<img width="784" alt="Screen Shot 2020-09-21 at 8 33 20 AM" src="https://user-images.githubusercontent.com/3514087/93788600-c9350880-fbe5-11ea-8cc1-0dbf71d1fbb1.png">

Reply:
This isn't quite accurate; `IntegerLiteralExprSyntax` is not limited to what can fit into the `Int` type; it's any sequence of characters that is syntactically a valid integer literal. The error you're showing occurs later on at the semantic analysis phase, but the compiler and SwiftSyntax will still parse it without error. If we try the following command:

```shell
echo 'let foo = 42424242424242424242424242424242424242424242424242424242424242424242424242' | \
  swiftc -frontend -emit-syntax -
```

we'll see this in the output:

```json
...
{
  "id":6,
  "kind":"IntegerLiteralExpr",
    "layout":[{
      "id":5,
      "tokenKind": {
        "kind":"integer_literal",
        "text":"42424242424242424242424242424242424242424242424242424242424242424242424242"
      },
```

So `potentialIntegerValue` called on this would return `nil`, and `integerValue` would crash.

Reply:
I agree with @allevato, this "overflows" error is not even coming from typechecking, it's coming from SIL code generation.
Many thanks to @xwu for catching this, I believe we need to make `integerValue` return an optional.

Reply:
Right, integer literals are untyped; they are _not_ “Int literals.”

Reply:
I [now](https://media.giphy.com/media/AuwBPJztsEWkw/giphy.gif) agree with @xwu's concern. Accordingly, I intend to create a story in the bug tracker and raise a follow-up PR.